packages/components/namespace-store/namespace-store-form.tsx

```diff
@@ -77,46 +81,65 @@ type NamespaceStoreFormProps = {
   onCancel: () => void;
 };
 
+const isFormValid = (form: NSSFormDataState) => {
+  const {
+    name,
+    accessKey,
+    secretKey,
+    secretName,
+    region,
+    endpoint,
+    target,
+    pvc,
+    pvcFolderName,
+  } = form;
+  return (
+    name &&
+    name.length <= 42 &&
+    (((region || endpoint) &&
+      (secretName || (secretKey && accessKey)) &&
+      target) ||
+      (pvc && pvcFolderName))
+  );
+};
+
 const NamespaceStoreForm: React.FC<NamespaceStoreFormProps> = withHandlePromise<
   NamespaceStoreFormProps & HandlePromiseProps
 >((props) => {
   const { t } = useTranslation();
-  const [nsName, setNsName] = React.useState('');
   const [provider, setProvider] = React.useState(BC_PROVIDERS.AWS);
-  const [pvc, setPVC] = React.useState(null);
-  const [folderName, setFolderName] = React.useState('');
-  const [providerDataState, providerDataDispatch] = React.useReducer(
-    providerDataReducer,
-    initialState
+  const [formDataState, formDataDispatch] = React.useReducer(
+    nssFormReducer,
+    initialStateFormCreation
```

Comment:
seems good that now we have state at one place... any reason why provider is not a part of reducer as well ??

Reply:
made provider part of reducer